README.md

```diff
@@ -1,170 +1,189 @@
-- **`websockets.py`**: WebSocket endpoints for global and conversation-specific event streaming
+The service acts as a pipeline, processing telemetry data and making it available for real-time consumption.
 
-#### Core Business Logic (`core/`)
-- **`span_preprocessor.py`**: Converts OpenTelemetry spans into typed events by extracting semantic attributes, mapping span kinds to event types, and preserving tracing context
-- **`connection_manager.py`**: Manages WebSocket connections with global broadcast and conversation-specific routing
-- **`state.py`**: Global application state management for connection managers and shared resources
+1.  **Agent Instruments**: Your agentic application is instrumented with OpenTelemetry to emit traces.
+2.  **Ingestion Endpoint**: The service receives these traces at its OTLP/HTTP endpoint (`/v1/traces`).
+3.  **Span Processing**: A core preprocessor (`span_preprocessor.py`) parses the trace spans, extracting key attributes and mapping them to structured event models (e.g., `AgentStart`, `LLMCall`, `ToolOutput`).
+4.  **WebSocket Broadcast**: The processed events are immediately broadcasted through a connection manager to all subscribed WebSocket clients.
 
-#### Data Models (`models/`)
-- **`events.py`**: Structured dataclasses for communication events (agent lifecycle, LLM calls, tool invocations)
-- **`content.py`**: Content type definitions for LLM requests/responses and tool interactions
-
-#### Utilities (`utils/`)
-- **`extractors.py`**: Attribute extraction logic for parsing OTLP span data
-- **`factories.py`**: Factory functions for creating specific event types from span data
+-----
 
 ## Prerequisites
 
-- **Python 3.13+**: Required for the app
-- **uv**: Package manager for Python projects
-- **Kubernetes cluster**: Required for Tilt development (see Local Kubernetes Setup below)
+To build, run, and contribute to this project, you'll need the following tools installed.
 
-## Setup
+| Tool | Version | Notes |
+| :--- | :--- | :--- |
+| **Python** | `>= 3.13` | The core programming language. |
+| **uv** | `latest` | A fast Python package installer and virtual environment manager. |
+| **Docker** | `>= 20.10+` | For containerized deployment. |
+| **pre-commit**| `latest` | Manages pre-commit hooks. Installed via `brew bundle`. |
+| **go-task** | `latest` | A modern task runner/build tool. Installed via `brew bundle`. |
+| **tilt** | `latest` | For local multi-service Kubernetes development. Installed via `brew bundle`.|
+| **Kubernetes**| `any` | Required for local development with Tilt. |
 
-### 1. Install Dependencies
+-----
 
-```bash
-# Install system dependencies
-brew bundle
+## Getting Started
 
-# Install Python dependencies
-uv sync
+Follow these steps to get the application running on your local machine.
 
-# Install frontend dependencies
-cd frontend && npm install && cd ..
-```
+### 1. Clone the Repository
```

Comment:
I'd assume people know that and probably done so already, so let's skip the clone step in Readmes.